src/main/java/org/mastodon/mamut/nearest/NearestObjectStatFeature.java

```diff
@@ -0,0 +1,291 @@
+package org.mastodon.mamut.nearest;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.mastodon.feature.Dimension;
+import org.mastodon.feature.Feature;
+import org.mastodon.feature.FeatureModel;
+import org.mastodon.feature.FeatureProjection;
+import org.mastodon.feature.FeatureProjectionKey;
+import org.mastodon.feature.FeatureProjectionSpec;
+import org.mastodon.feature.FeatureSpec;
+import org.mastodon.feature.Multiplicity;
+import org.mastodon.mamut.model.Model;
+import org.mastodon.mamut.model.Spot;
+import org.mastodon.mamut.nearest.NearestObjectStatModel.CollectBy;
+import org.mastodon.mamut.nearest.NearestObjectStatModel.NearestObjectStatItem;
+import org.mastodon.properties.DoublePropertyMap;
+import org.mastodon.ui.coloring.feature.FeatureProjectionId;
+
+public class NearestObjectStatFeature implements Feature< Spot >
+{
+
+	public static final String KEY = "Stats on nearest neighbors";
+
+	private static final String HELP_STRING =
+			"Stores the values of the statistics on the N nearest neighbors plugin.";
+
+	private final Map< FeatureProjectionKey, MyDoublePropertyProjection > featureProjections;
+
+	private final Map< NearestObjectStatItem, FeatureProjectionKey > fSpecMap;
```

Comment:
Consider renaming this variable as it is difficult to identify its meaning from its name